sum.go

```diff
@@ -31,6 +41,23 @@ func Sum(data []byte, code int, length int) (Multihash, error) {
 		d, err = sumSHA3(data)
 	case DBL_SHA2_256:
 		d = sumSHA256(sumSHA256(data))
+	case BLAKE2B:
+		switch length {
+		case 32:
+			out := blake2b.Sum256(data)
+			d = out[:]
+		case 48:
+			out := blake2b.Sum384(data)
+			d = out[:]
+		case 64:
+			out := blake2b.Sum512(data)
+			d = out[:]
+		default:
+			return nil, fmt.Errorf("unsupported length for blake2b")
+		}
```

Comment:
--- btw, i think i understand @whyrusleeping's desire a bit more now, it sounds like what he wants is:

- if the function has a length input defined, and the function designers call for "256 bit blake2b hash" always being X (i.e. always recomputed shorter, never truncated), then it may be worth the wide-spread confusion and attempt to educate people NEVER to truncate, ALWAYS to recompute shorter. it would also require writing libraries that allow _any_ byte length (the Go one does not, and only allows `256, 384, 512`)-- otherwise people will just truncate.
- note this is _certainly not_ what should happen to sha2, sha3, given that people routinely use larger functions and truncate them. it is an explicit goal of multihash to make that madness easier to deal with.

---

So I claim there is a binary choice of "truncate is what people do in practice" vs "recompute is what people do in practice". anywhere in the middle is confusing, problematic, and hard to deal with. Sha-family is squarely in "truncate is what people do in practice", simply by what happens in practice, even though arguably we should work hard to be in "recompute, ...". If the claim is that blake2b IS squarely in "recompute..." and that people indeed do not truncate, always recompute, then that's exciting. but im also not sure it can stay there long term without a lot of constant work to overturn how people tend to use hashes.